### PR TITLE
Create ORM

### LINKED ISSUE #6: Develop ORM
We need to build out the entire class system modeling the SQL table structure that our `.sql` file creates using the SQLAlchemy ORM. 

I would advocate for putting all the models in some `music2/common/models` directory or something. We can nest them into further subdirectories and keep related things in the same file, but whatever we do we just need to keep in mind that these classes will end up having extra business logic methods tacked on specific to each model, so these classes will probably need some room to grow. Alternatively, we could just create utility functions that take instances of these classes to perform our business logic so that we can keep these classes as light as possible, but we will definitely end up having extra convenience methods for specific queries and db-related actions.

Once this is done, we should be able to use alembic to create a migration to sync our actual SQL structure with the constructed classes and it should be empty.

### PR DESCRIPTION
**Pull request by alatar224**
Creates a ton of ORM classes and adds alembic as a dependency. I autogenerated an initial alembic revision comparing the db state after running the `music_pg.sql` script and the one specificed by the model classes. Here is a list of all the differences:

Create indexes on foreign key columns:
```
  artist_primary_artist_id_idx
  artist_genre_artist_id_idx
  artist_genre_genre_id_idx
  bracket_selection_bracket_id_idx
  bracket_selection_project_id_idx
  match_next_match_id_idx
  match_winner_id_idx
  match_seed_match_id_idx
  match_seed_previous_match_id_idx
  match_seed_seed_id_idx
  notification_participant_id_idx
  notification_submission_id_idx
  seed_bracket_id_idx
  song_artist_artist_id_idx
  song_artist_song_id_idx
  song_genre_genre_id_idx
  song_genre_song_id_idx
  submission_bracket_id_idx
  submission_participant_id_idx
  submission_song_id_idx
```

Set ON DELETE behavior of foreign keys:
```
  artist_primary_artist_id_fkey RESTRICT
  artist_genre_artist_id_fkey CASCADE
  artist_genre_genre_id_fkey CASCADE
  bracket_selection_project_id_fkey CASCADE
  bracket_selection_bracket_id_fkey CASCADE
  config_project_id_fkey CASCADE
  genre_relation_parent_genre_id_fkey SET NULL
  genre_relation_child_genre_id_fkey SET NULL
  link_relation_parent_link_id_fkey CASCADE
  link_relation_child_link_id_fkey CASCADE
  match_next_match_id_fkey SET NULL
  match_winner_id_fkey SET NULL
  match_round_id_fkey CASCADE
  match_seed_seed_id_fkey CASCADE
  match_seed_previous_match_id_fkey SET NULL
  match_seed_match_id_fkey CASCADE
  notification_participant_id_fkey CASCADE
  notification_submission_id_fkey SET NULL
  participant_nickname_participant_id_fkey CASCADE
  project_project_lead_id_fkey SET NULL
  round_bracket_id_fkey CASCADE
  seed_bracket_id_fkey CASCADE
  seed_submission_id_fkey CASCADE
  seed_history_seed_id_fkey CASCADE
  seed_history_submission_id_fkey CASCADE
  song_artist_artist_id_fkey CASCADE
  song_artist_song_id_fkey CASCADE
  song_genre_song_id_fkey CASCADE
  song_genre_genre_id_fkey CASCADE
  song_relation_parent_song_id_fkey CASCADE
  song_relation_child_song_id_fkey CASCADE
  submission_participant_id_fkey CASCADE
  submission_bracket_id_fkey CASCADE
  submission_song_id_fkey CASCADE
  submission_history_bracket_id_fkey CASCADE
  submission_history_submission_id_fkey CASCADE
  submission_history_participant_id_fkey CASCADE
  submission_history_song_id_fkey CASCADE
```

Split unique constraint indexes by column (all except seed.seed are FK):
```
  artist_genre_artist_id_genre_id_key
  seed_bracket_id_seed_key
  song_artist_song_id_artist_id_key
  song_genre_song_id_genre_id_key
```

Create index on unique constraint columns:
```
  seed_seed_idx
```

Drop columns:
```
  genre.parent_id
```

Make columns nullable:
```
  genre_relation.parent_genre_id
  genre_relation.child_genre_id
  match.date_posted
```

Make columns non-nullable:
```
  match.tie
  match_seed.votes
  project.name
  seed.emoji
  seed_history.emoji
```

Other:
* `match.tie` - change from `text` to `int`
* `notification.date_succeeded` - rename from `date_succeded`
* `participant_nickname.date_start` - change from `date` to `datetime`
* `participant_nickname.date_end` - change from `date` to `datetime`
* `round.index_desc` - change comment from `'0-indexed from finals'` to `'0-indexed from Finals'`
* `seed_history.update_date` - rename from `date_changed`
* `seed_history.update_note` - rename from `reason_changed`
* `submission.compsers` - rename from `composer`
* `submission_history.composers` - rename from `composer`

### CODE REVIEW

File: music2/models/round.py
Comment (yorkshirelandscape) [inline music2/models/round.py:1]: I've always considered the Finals and the Third Place Match to be separate rounds. Not sure how you want to address it.

File: music2/models/song.py
Comment (yorkshirelandscape) [inline music2/models/song.py:1]: Let's add 'alternate' as a song_relation_type.

**Review by yorkshirelandscape — APPROVED** [review]
I've added a couple comments with improvements, but this all looks good.

File: music2/models/round.py
Comment (alatar224) [reply music2/models/round.py:1]: I'd be fine either way really, whatever's easiest to set up on the backend is what gets my preference probably. I'll go ahead and leave this in for now and when we actually get to trying to add in the test data and seeing how the third place match fits into that we can adjust.

File: music2/models/song.py
Comment (alatar224) [reply music2/models/song.py:1]: Done

**Pull request merged**